src-plugins/itkFilters/itkFiltersSubtractProcess.cpp

```diff
@@ -91,20 +91,21 @@ int itkFiltersSubtractProcess::updateProcess(medAbstractData* inputData)
 {
     typename ImageType::Pointer inputImage = static_cast<ImageType*>(inputData->data());
 
-    typedef itk::SubtractImageFilter< ImageType, itk::Image<double, ImageType::ImageDimension>, ImageType >  SubtractFilterType;
-    typename SubtractFilterType::Pointer subtractFilter = SubtractFilterType::New();
-
-    subtractFilter->SetInput(inputImage);
-    subtractFilter->SetConstant ( d->subtractValue );
+    using ShiftScaleFilterType = itk::ShiftScaleImageFilter<ImageType, ImageType >;
+    typename ShiftScaleFilterType::Pointer shiftFilter = ShiftScaleFilterType::New();
+    shiftFilter->SetInput( inputImage );
+    shiftFilter->SetScale( 1.0 );
+    // Shift in the limit of the image type
+    double negValue = -1.0 * d->subtractValue;
+    shiftFilter->SetShift( negValue );
+    shiftFilter->Update();
 
     itk::CStyleCommand::Pointer callback = itk::CStyleCommand::New();
     callback->SetClientData ( ( void * ) this );
     callback->SetCallback ( itkFiltersProcessBase::eventCallback );
-    subtractFilter->AddObserver ( itk::ProgressEvent(), callback );
-
-    subtractFilter->Update();
+    shiftFilter->AddObserver ( itk::ProgressEvent(), callback );
```

Comment:
This line should be before the call to update because the calculation will already be finished by this point.

Reply:
Done